matrix-sdk/src/main/java/org/matrix/androidsdk/RestClient.java

```diff
@@ -190,6 +190,7 @@ public Response intercept(Chain chain) throws IOException {
             Pair<SSLSocketFactory, X509TrustManager> pair = CertUtil.newPinnedSSLSocketFactory(hsConfig);
             okHttpClientBuilder.sslSocketFactory(pair.first, pair.second);
             okHttpClientBuilder.hostnameVerifier(CertUtil.newHostnameVerifier(hsConfig));
+            okHttpClientBuilder.connectionSpecs(CertUtil.newConnectionSpecs(hsConfig));
```

Comment:
This code will be called even if no particular settings in hsConfig. It will override the default value, set to  OkHttpClient.DEFAULT_CONNECTION_SPECS. Is it OK?

Reply:
No, it will be MODERN_TLS with all TLS versions and cipher suites available.
The DEFAULT_CONNECTION_SPECS allow MODERN_TLS (without specifying any TLS versions and cipher suites) AND cleartext (unencrypted connections).

I change the code for wrt TLS versions and cipher suites (i.e. I will not explicitely allow all TLS versions and cipher suites) but for the cleartext I don't think it a good idea. Is there any unencrypted HTTP connection allowed to the home server ?

Reply:
ok, I was wondering about the `cleartext` case, but I check internally and we only to https connection.

Reply:
FTR, the crypto tests use http connection... so fails

Reply:
Could clear text connections be re-enabled? Requiring HTTPS prevents the use of a private homeserver through a VPN for example. See https://github.com/vector-im/riot-android/issues/2495.